src/storage/src/hummock/sstable_store.rs

```diff
@@ -267,49 +273,163 @@ impl SstableStore {
             .map_err(Into::into)
     }
 
-    pub async fn preload_blocks(
+    pub async fn get_stream(
         &self,
         sst: &Sstable,
         start_index: usize,
         mut end_index: usize,
-    ) -> HummockResult<Option<BatchBlockStream>> {
+        policy: CachePolicy,
+        long_connection_prefetch: bool,
+        stats: &mut StoreLocalStatistic,
+    ) -> HummockResult<Box<dyn BlockStream>> {
         let object_id = sst.id;
-        if self.block_cache.exists_block(object_id, start_index as u64) {
-            return Ok(None);
+        let pending_data = self.prefetch_buffer_usage.load(Ordering::SeqCst);
+        if pending_data > self.prefetch_buffer_capacity || !long_connection_prefetch {
+            return self
+                .prefetch_blocks(sst, start_index, end_index, policy, stats)
+                .await;
         }
-        let start_offset = sst.meta.block_metas[start_index].offset as usize;
-        let mut end_offset = start_offset;
+        if let Some(holder) = self.block_cache.get(object_id, start_index as u64) {
+            return Ok(Box::new(PrefetchBlockStream::new(
+                VecDeque::from([holder]),
+                start_index,
+            )));
+        }
+        let mut data_size = 0;
         for idx in start_index..end_index {
-            if self.block_cache.exists_block(object_id, idx as u64) {
+            if idx > start_index && self.block_cache.exists_block(object_id, idx as u64) {
                 end_index = idx;
                 break;
             }
-            end_offset += sst.meta.block_metas[idx].len as usize;
+            data_size += sst.meta.block_metas[idx].len as usize;
         }
-        let pending_data = self.pending_streaming_loading.load(Ordering::SeqCst);
-        if end_offset == start_offset || pending_data > self.large_query_memory_usage {
-            return Ok(None);
-        }
-        let data_path = self.get_sst_data_path(object_id);
-        let memory_usage = std::cmp::min(end_offset - start_offset, self.store.recv_buffer_size());
-        self.pending_streaming_loading
-            .fetch_add(memory_usage, Ordering::SeqCst);
-        let reader = self
-            .store
-            .streaming_read(&data_path, start_offset..end_offset)
+
+        let inner = self
```

Comment:
Shall we also maintain `cache_data_prefetch_count` and `cache_data_prefetch_block_count` when `LongConnectionBlockStream` is used?